sdk/core/Azure.Core/tests/TestFramework/RecordMatcher.cs

```diff
@@ -77,6 +86,23 @@ public virtual RecordEntry FindMatch(Request request, IList<RecordEntry> entries
             throw new InvalidOperationException(GenerateException(request.Method, uri, headers, bestScoreEntry));
         }
 
+        public virtual bool IsEquivalentResponse(RecordEntry entry, RecordEntry otherEntry)
+        {
+            var headeders1 = entry.ResponseHeaders.Where(h => !ExcludeResponseHeaders.Contains(h.Key)).ToArray();
+            var headers2 = otherEntry.ResponseHeaders.Where(h => !ExcludeResponseHeaders.Contains(h.Key)).ToArray();
+            return
+                entry.StatusCode == otherEntry.StatusCode &&
+                headeders1.SequenceEqual(headers2, new HeaderComparer()) &&
+                IsBodyEquivalent(
+                    entry.ResponseBody ?? Array.Empty<byte>(),
+                    otherEntry.ResponseBody ?? Array.Empty<byte>());
+        }
+
+        protected virtual bool IsBodyEquivalent(byte[] body, byte[] otherBody)
```

Comment:
Can we pass both of the `RecordEntry` objects instead of just their bodies?  I'm fine with the user checking for `null` before analyzing `ResponseBody`.  The Storage tests call multiple services with a mix of JSON/XML response bodies and it'll be easier to figure out how to compare if I've got the full record.